tests/test_qp.py

```diff
@@ -22,317 +24,245 @@ class QPTest(PyverbsAPITestCase):
     Test various functionalities of the QP class.
     """
 
-    def test_create_qp_no_attr_connected(self):
-        """
-        Test QP creation via ibv_create_qp without a QPAttr object proivded.
-        Checked QP types are RC and UC.
-        """
-        for ctx, attr, attr_ex in self.devices:
-            with PD(ctx) as pd:
-                with CQ(ctx, 100, None, None, 0) as cq:
-                    qia = get_qp_init_attr(cq, attr)
-                    qia.qp_type = e.IBV_QPT_RC
-                    try:
-                        with QP(pd, qia) as qp:
-                            assert qp.qp_state == e.IBV_QPS_RESET, 'RC QP should have been in RESET'
-                    except PyverbsRDMAError as ex:
-                        if ex.error_code == errno.EOPNOTSUPP:
-                            raise unittest.SkipTest('Create QP with RC attrs is not supported')
-                        raise ex
-                    qia.qp_type = e.IBV_QPT_UC
-                    try:
-                        with QP(pd, qia) as qp:
-                            assert qp.qp_state == e.IBV_QPS_RESET, 'UC QP should have been in RESET'
-                    except PyverbsRDMAError as ex:
-                        if ex.error_code == errno.EOPNOTSUPP:
-                            raise unittest.SkipTest('Create QP with UC attrs is not supported')
-                        raise ex
-
-
-    def test_create_qp_no_attr(self):
-        """
-        Test QP creation via ibv_create_qp without a QPAttr object proivded.
-        Checked QP types are Raw Packet and UD. Raw Packet is skipped for
-        non-root users / Infiniband link layer.
-        """
+    def create_qp(self, creator, qp_init_attr, is_ex, with_attr):
+        try:
+            qp_attr = (None, QPAttr())[with_attr]
+            return QP(creator, qp_init_attr, qp_attr)
+        except PyverbsRDMAError as ex:
+            if ex.error_code == errno.EOPNOTSUPP:
+                with_str = ('with', 'without')[with_attr] + ('', ' extended')[is_ex]
+                raise unittest.SkipTest(f'Create {get_qp_type_str(qp_init_attr.qp_type)} QP {with_str} attrs is not supported')
```

Comment:
please assign "get_qp_type_str(qp_init_attr.qp_type)" to a temp var. (128c length is just too much)

Reply:
Done.